### PR TITLE
Chore/287 document aspnet api routing

### LINKED ISSUE #287: Document ASP.NET API routing
To prevent knowledge silos and minimize the bus factor we should document the routing of our asp.net api.

### PR DESCRIPTION
**Pull request by myieye**
Resolves #287 

A general update to our README's including documentation on the various routes exposed by our API as well as our numerous hostnames and what they route to.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,56 +47,58 @@ If you're running Windows, you may need to add the following lines to your `C:\W
...
 
 ---
-### Helpful urls
-* http://localhost - k8s ingress
-* http://localhost:3000 - sveltekit frontend
-* http://localhost:5158/api/swagger - swagger docs for the api
-* http://localhost:5158/api/graphql/ui - graphiql UI
-* http://localhost:5158/api/graphql - graphiql endpoint
+### Project urls
+* http://localhost - k8s ingress / SvelteKit UI
```
Comment (hahn-kev) [inline README.md:97]: if you're going to mention sveltekit here then the api should also be mentioned. That said everything goes to ingress which routes it to the correct location

File: backend/README.md [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
-The backend is a dotnet core api. With a proxy to route requests for hg and resumable.
\ No newline at end of file
+The LexBox backend is a dotnet core api. With a proxy to route requests for hg and resumable.
+
+### ASP.NET Routes
+
+| Route | Description | Notes |
+|---|---|---|
+| **Send/Receive** | |
+| `/{project-code}`<sup>1</sup> OR `/hg/{project-code}` | hg S/R  | <sup>1</sup>Obviously it would be a bit problematic if a `project-code` were `api`, but that's the URL baked into Chorus clients around the world 🤷. (Do we still need this or could our proxies send our hg domains to `/hg`?) |
```
Comment (hahn-kev) [inline backend/README.md:10]: sadly no, ingress has very limited support for rewriting requests which is what that would require.

File: backend/README.md [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
+| `/api/v03` | hg-resumable S/R |   |
+| `/api/user/{userName}/projects` | Returns a list of the user's projects  |  See `LegacyProjectApiController`. <br>Used by [Chorus/FLEx](https://github.com/sillsdev/chorus/blob/04eda7903f3fe09d92cfc3edf91acea598c92744/src/LibChorus/Model/ServerSettingsModel.cs#L336) & [Language Forge](https://github.com/sillsdev/web-languageforge/blob/b2123ad2ca45a67bbd68381152e98b6f2bb5334a/src/Api/Model/Languageforge/Lexicon/Command/SendReceiveCommands.php#L91). |
+| **Frontend API** | |
+| `/api/graphql` | GraphQL API |  |
+| `/api/project/upload-zip/{project-code}` | TUS upload  | Expects a zip with an .hg folder  |
+| `/api/**` | Other REST controllers  |  |
+| **Tools** | |
+| `/api/healthz` | Health check  |
+| `/api/graphql/ui` | GraphQL UI | [Banana Cake Pop](https://chillicream.com/docs/bananacakepop/v2/explore-the-ui) |
+| `/api/quartz`  | Quartz.NET UI (job scheduler)  | [CrystalQuartz](https://github.com/guryanovev/CrystalQuartz) |
```
Comment (hahn-kev) [inline backend/README.md:20]: should add `/api/swagger`

Comment (hahn-kev) [reply backend/README.md:20]: and `/security.txt` and `/.well-known/security.txt`

File: backend/README.md
Comment (hahn-kev) [inline backend/README.md:1]: there is an order of precedance for requests, which is not represented in the order shown here. Namly S&R endpoints are lowest precedance. Is that useful here or not?

File: deployment/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  - hg
+  - hg-resumable
+- ui (SvelteKit app)
+
+### Local development
+
+Same as above plus:
+
+- lexbox
+  - maildev
```
Comment (hahn-kev) [inline deployment/README.md:21]: said db-migrations init container is removed in local dev, migrations are done at app startup there.

File: deployment/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+### Production, Staging & Develop
+
+- lexbox
+  - .net API (GQL API, REST API, hg proxy etc. (see [backend README](../backend/README.md)))
+  - OTEL collector
+- db (PostgresDB)
+- hg
+  - hg
+  - hg-resumable
```
Comment (hahn-kev) [inline deployment/README.md:13]: everywhere except production there's a container called `populate-test-repos` that will download repos at startup

File: backend/README.md
Comment (myieye) [reply backend/README.md:1]: I prefer having them grouped by type and ordered by importance. Either way, I don't think it's too critical. I'll add a note about their precedence and then see what you think.

File: backend/README.md [hunk 6]
```diff
@@ -1,4 +1,22 @@
...
+| **Frontend API** | |
+| `/api/graphql` | GraphQL API |  |
+| `/api/project/upload-zip/{project-code}` | TUS upload  | Expects a zip with an .hg folder  |
+| `/api/**` | Other REST controllers  |  |
+| **Tools** | |
+| `/api/healthz` | Health check  |
+| `/api/swagger` | Swagger UI | |
+| `/api/graphql/ui` | GraphQL UI | [Banana Cake Pop](https://chillicream.com/docs/bananacakepop/v2/explore-the-ui) |
+| `/api/quartz`  | Quartz.NET UI (job scheduler)  | [CrystalQuartz](https://github.com/guryanovev/CrystalQuartz) |
+| `/api/security.txt` or `/api/.well-known/security.txt`  | Vulnerability reporting info  | [Security.txt standard](https://securitytxt.org/) |
```
Comment (hahn-kev) [inline backend/README.md:22]: actuall niether of these are prefixed with `/api` to follow the standard.
```suggestion
| `/security.txt` or `/.well-known/security.txt`  | Vulnerability reporting info  | [Security.txt standard](https://securitytxt.org/) |
```

**Review by hahn-kev — CHANGES_REQUESTED** [review]
Nice job! I appreciate the links to docs for stuff like Securitytxt and some of the UI backends.

File: backend/README.md (hunk 6 above)
Comment (myieye) [reply backend/README.md:22]: Oops!!

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**